### PR TITLE
PCHR-3742: Disabling or Deleting Options Breaks Job Roles Tab

### PR DESCRIPTION
**Pull request by ajesamson**
## Overview
This PR fixes issues with editing disabled or deleted job role option group. Dropdown displays wrongly if such record was previously submitted.

## Before
before_roles

## After
after_roles

## Technical Details
Values that are invalid gets submitted as the existing implementation only checks if the value is defined or not. To guide against this, submitted data is checked against valid list of options data.

```js
updateLocation = _.includes(_.keys(vm.LocationsData), updatedRole.location);
updateLevel = _.includes(_.keys(vm.LevelsData), updatedRole.level);
updateDepartment = _.includes(_.keys(vm.DepartmentsData), updatedRole.department);
updateRegion = _.includes(_.keys(vm.RegionsData), updatedRole.region);

updatedRole.location = (updatedRole.location === undefined || !updateLocation) ? 
  updatedRole.location = '' : updatedRole.location;
updatedRole.level = (updatedRole.level === undefined || !updateLevel) ? updatedRole.level = '' : 
  updatedRole.level;
updatedRole.department = (updatedRole.department === undefined || !updateDepartment) ? 
  updatedRole.department = '' : updatedRole.department;
updatedRole.region = (updatedRole.region === undefined || !updateRegion) ? updatedRole.region = 
  ''  : updatedRole.region;
```

---
✅Manual Tests - passed
✅Jasmine Tests - passed
✅JS distributive files - included

### CODE REVIEW

File: com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js [hunk 1]
```diff
@@ -1122,10 +1122,16 @@ define([
       }
 
       updatedRole = angular.copy(vm.editData[roleId]);
-      updatedRole.location = (updatedRole.location === undefined) ? updatedRole.location = '' : updatedRole.location;
-      updatedRole.level = (updatedRole.level === undefined) ? updatedRole.level = '' : updatedRole.level;
-      updatedRole.department = (updatedRole.department === undefined) ? updatedRole.department = '' : updatedRole.department;
-      updatedRole.region = (updatedRole.region === undefined) ? updatedRole.region = '' : updatedRole.region;
+      // Ensure location, level, department, region exist and not disabled
+      updateLocation = _.includes(_.keys(vm.LocationsData), updatedRole.location);
```
Comment (igorpavlov) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js]: Please move the code to a separate function with a meaningful name, there is too much duplication.

File: com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js [hunk 2]
```diff
@@ -1122,10 +1122,16 @@ define([
...
+      updateLevel = _.includes(_.keys(vm.LevelsData), updatedRole.level);
+      updateDepartment = _.includes(_.keys(vm.DepartmentsData), updatedRole.department);
+      updateRegion = _.includes(_.keys(vm.RegionsData), updatedRole.region);
+
+      updatedRole.location = (updatedRole.location === undefined || !updateLocation) ? updatedRole.location = '' : updatedRole.location;
```
Comment (igorpavlov) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js]: Please move the code to a separate function with a meaningful name, there is too much duplication.

**Comment by igorpavlov** [comment]
@ajesamson are there JS test suites for JobRoles extension?

**Comment by ajesamson** [comment]
@igorpavlov There are JS test suites.

File: com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js [hunk 3]
```diff
@@ -1146,6 +1146,30 @@ define([
       });
     }
 
+    /**
+     * Update submitted role if undefined or option type is disabled
+     *
+     * @param updatedRole
```
Comment (igorpavlov) [inline com.civicrm.hrjobroles/js/src/job-roles/controllers/job-roles.controller.js]:
```js
* @param  updatedRole
* @param  optionData
* @return {Object}
```

**Pull request merged**